text/0000-formatted-bliss.md

```diff
@@ -0,0 +1,226 @@
+
+Eventually, when the entire repository has been formatted,
+the `.enforcefmt` trigger may be removed in favor of checking everything.
+
+## 3. Add the `./x.py fmt` command
+
+A command `fmt` is added to rustbuild (`./x.py`).
+The command will format the repository according to the style
+but only in folders with `.enforcefmt` files as described above.
+
+## 4. Add a `@rustbot fmt` command
+
+Sometimes compiler developers do not have the time, or may simply forget,
+to apply `./x.py fmt`. In other situations, the developer may not have a
+computer with a compiler and git handy. It may also happen that the PR
+is perfectly good aside from formatting. For these situations, a
+`@rustbot fmt` command is provided with which the author, a reviewer,
+or a T-release member can simply ask `@rustbot` to run `./x.py fmt`
+on their PR and add a commit for that.
+
+## 5. Add the `./x.py install-fmt-hook`
+
+We add a command `install-fmt-hook` to rustbuild.
+This command will install a pre-commit git-hook.
+This git-hook will check that the formatting in the commit is good.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There are two notable drawbacks to this proposal:
+
+1. Formatting commits will need to be added to the repository.
+   This will be in the way of checking who is to `git blame`
+   for a certain change to the compiler. We should try to
+   limit the cost of this to as one-time as possible.
+
+   Other than that, it is unavoidable to have one such formatting
+   commit in the entire repository if we wish to enforce a style.
+   Moreover, people do drive-by commits for style changes sometimes
+   so this cost already exists to some extent. By enforcing style
+   mechanically, we will have fewer such PRs.
+
+2. More impediments to merging a PR.
+
+   Since we now mechanically check and enforce a style,
+   some developers may feel an additional burden when hacking on the compiler.
+
+   Moreover, automatic enforcement of style may make rollups harder to land
+   as a PR which would otherwise be guaranteed to merge may now fail due to
+   incorrect formatting. To mitigate that risk and to not overburden
+   the `bors` queue more than it is today, it is important that CI will check
+   for style problems early on.
+
+   In both cases, the `@rustbot fmt` command should help reduce the burden
+   of automatic style checking.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## The proposed style
+
+[small]: https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#use_small_heuristics
+[churn]: https://internals.rust-lang.org/t/running-rustfmt-on-rust-lang-rust-and-other-rust-lang-repositories/8732/81
+
+As aforementioned, the standard Rust Style Guide is adopted.
+This style is adopted to make the repository feel idiomatic in terms of style.
+
+However, we make a small modification to the style guide.
+In particular, we enable [`use_small_heuristics = "Max"`][small].
```

Comment:
FWIW this, `newline_style = "Unix"`, and `edition = "2018"` are the only rustfmt.toml sections modifications that Fuchsia uses as well.